osquery/database/database_plugin.cpp

```diff
@@ -0,0 +1,92 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <osquery/database_plugin.h>
+#include <osquery/logger.h>
+#include <osquery/registry_factory.h>
+#include <osquery/status.h>
+
+#include <osquery/flags.h>
+
+namespace osquery {
+
+Status DatabasePlugin::reset() {
+  // Keep this simple, scope the critical section to the broader methods.
+  tearDown();
+  return setUp();
+}
+
+Status DatabasePlugin::scan(const std::string& domain,
+                            std::vector<std::string>& results,
+                            const std::string& prefix,
+                            size_t max) const {
+  return Status(0, "Not used");
+}
+
+Status DatabasePlugin::call(const PluginRequest& request,
+                            PluginResponse& response) {
+  if (request.count("action") == 0) {
+    return Status(1, "Database plugin must include a request action");
+  }
+
+  // Get a domain/key, which are used for most database plugin actions.
+  auto domain = (request.count("domain") > 0) ? request.at("domain") : "";
+  auto key = (request.count("key") > 0) ? request.at("key") : "";
```

Comment:
This is a double search for the same key in each map. It could be made with one. Smth like:
```c++
const auto it = request.find("domain");
auto domain = it != request.end() ? it->second : "";
``` 
What do you think?

Reply:
(Unrelated to this PR) this practice of (1) search for existence and then (2) search again and access is used everywhere. There’s a fair amount of across-codebase aggregated performance wins if someone took a day and move them to find.